### ISSUE TITLE
[BUG] `NCFModel` throws an error when used with `mm.RegressionTask`

### ISSUE THREAD

**Issue by radekosmulski** [NVIDIA-Merlin/models#655]
### Bug description

An assert gets triggered which doesn't seem to apply to a regression task and stops the training.
![image](https://user-images.githubusercontent.com/2444926/184756263-2aed352f-3e32-4c63-8513-cf8c15339337.png)

### Steps/Code to reproduce bug
```
import os
import numpy as np

from nvtabular.loader.tf_utils import configure_tensorflow

configure_tensorflow()

import nvtabular as nvt
from nvtabular.ops import *
from merlin.models.utils.example_utils import workflow_fit_transform, save_results

from merlin.schema.tags import Tags

import merlin.models.tf as mm
from merlin.io.dataset import Dataset

import tensorflow as tf

from merlin.datasets.entertainment import get_movielens

input_path = os.environ.get("INPUT_DATA_DIR", os.path.expanduser("~/merlin-framework/movielens/"))
get_movielens(variant="ml-1m", path=input_path);
train, valid = get_movielens(variant="ml-1m", path=input_path);

model = mm.benchmark.NCFModel(
    train.schema.select_by_name(['userId', 'movieId', 'rating']),
    embedding_dim=64,
    mlp_block=mm.MLPBlock([128, 64]),
    prediction_tasks=mm.RegressionTask('rating'),
)

batch_size = 16 * 1024
LR = 0.03

opt = tf.keras.optimizers.Adagrad(learning_rate=LR)
model.compile(optimizer=opt, run_eagerly=False, metrics=[tf.keras.metrics.AUC()])
model.fit(train, validation_data=valid, batch_size=batch_size)
```

### Expected behavior
Training runs successfully. You can train a regression model with NCF.

### Environment details
current main on merlin-models

**Comment by rnyak** [comment]
@radekosmulski do you mind to give more context for what's the motivation to use `AUC()` metric with a regression task? Would RMSE or MSE be not more proper metric for this task?

**Comment by radekosmulski** [comment]
@rnyak I missed that AUC was being used there, I just copied the code 🙂 But I also suspect this issue is not related to the metric being used, there is an error earlier in the model stack, AFAICT. There was a discussion on slack where I think I understand the situation a bit better now -- that these tasks are not meant to be interchangeable. If we want to move from classification to regression then we might need to modify the base architecture a little bit.

Will close this issue as I don't believe it as a bug anymore.
